### DISCUSSION TITLE
AGH not able to resolve local hostnames

### DISCUSSION THREAD

**Discussion by bobloadmire** [AdguardTeam/AdGuardHome#6928] [Q&A]
I recently added adguard home, and using OPNsense as my router. For some reason AGH only shows local IPs, no hostnames. I tried enabling private reverse dns lookup by pointing it to OPNsense which has unbound by default, didn't make a difference. Is there a setting I'm missing in AGH, or maybe something needs to be enabled in OPNsense? Thanks.

Upstream DNS:
1.1.1.1
8.8.4.4
[/in-addr.arpa/]192.168.0.1
[/ip6.arpa/]192.168.0.1

Private reverse DNS servers:
192.168.0.1 (Unbound on OPNsense)

Use Private Reverse DNS Resolvers enabled, enable reverse resolving of clients IP address enabled.

**Comment by schneich** [comment]
Same issue here. I am using Adguard Home v.0.107.48.

DNS settings:
![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/84641760/4180ce04-0cab-4e2e-bfab-8eff24c9c19d)

![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/84641760/cab55a68-ef88-4a50-9e2a-7802c5b04155)


Runtime clients:
Within Adguard Home, there is a list of Runtime clients. There are some devices there, but this is an old, not updated list. How do I force Adguard Home to update this?
![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/84641760/8494b104-1a9a-4dbe-8986-91210ec1849d)



Or is there a setting in my Unifi Network Application to enable rDNS?

**Comment by schneich** [comment]
Hi @bobloadmire ,

I spent a couple of nights in this.

With these settings, reverse-DNS is working perfectly fine.
Upstream DNS servers
`[/168.192.in-addr.arpa/]192.168.1.1
[/internal/]192.168.1.1
https://dns.quad9.net/dns-query`

Private reverse DNS servers
`192.168.1.1`

The issue was not AdGuard Home, but a messy `/etc/hosts`-file on my Unifi USG-3P Gateway. After deleting all the entries in that file, the gateway started to gather all the data about my clients again and now they show the correct hostname and domain within AdGuard Home.
Details:
- https://community.ui.com/questions/resolve-domain-of-docker-container-docker-network-ipvlan/c3848c1e-5489-4b61-a899-8089db7444b3
- https://forums.docker.com/t/how-to-address-docker-container-via-hostname-ipvlan-adguard-unifi/140973/1

Perhaps, this helps you as well.

Chris

**Comment by BoBBer446** [comment]
I was going to search a bit an i found this:
AdGuardHome.yaml add this:
`resolve_clients: true` and it was working. I see Hostnames. But im not shure why i cant set it up on webinterface. "Resolve Hostanmes" was not working.

```
dns:
  bind_host: 0.0.0.0
  port: 53
  resolve_clients: true
  use_private_ptr_resolvers: true
  local_ptr_upstreams:
    - 192.168.178.1
upstream_dns:
  - https://dns10.quad9.net/dns-query
  - '[\/fritz\.box\/]192.168.178.1'
```

**Comment by mcarver** [comment]
Hello I'm troubleshooting a similar problem with my clients resolving and noticed that `resolve_clients: true` option in `AdGuardHome.yaml` doesn't exist at all, even though the box is checked in the UI.  Unchecking and saving doesn't add the option to the file.

I documented it in another issue here:  https://github.com/AdguardTeam/AdGuardHome/issues/7586#issuecomment-2626110431

**Comment by kevinholtkamp** [comment]
I just had the same issue and for me, the solution @BoBBer446 provided worked perfectly
